frontend/src/scenes/feature-flags/featureFlagLogic.ts

```diff
@@ -404,7 +404,21 @@ export const featureFlagLogic = kea<featureFlagLogicType>([
                     if (!state) {
                         return state
                     }
-                    return { ...state, filters: { ...state.filters, multivariate: multivariateOptions } }
+                    const variantsSet = new Set(multivariateOptions?.variants.map((variant) => variant.key))
+                    const groups = state.filters.groups.map((group) =>
+                        !group.variant || variantsSet.has(group.variant) ? group : { ...group, variant: null }
+                    )
+                    const oldPayloads = state.filters.payloads ?? {}
+                    const payloads = Object.keys(oldPayloads).reduce((newPayloads, variantKey) => {
+                        if (variantsSet.has(variantKey)) {
+                            return { ...newPayloads, [variantKey]: oldPayloads[variantKey] }
+                        }
+                        return newPayloads
+                    }, {})
```

Comment:
This does an object spread on every iteration. We can make it more performant.

```suggestion
                    const payloads: Record<string, JsonType> = {}
                    for (const variantKey of Object.keys(oldPayloads)) {
                        if (variantsSet.has(variantKey)) {
                            payloads[variantKey] = oldPayloads[variantKey]
                        }
                    }
```